App/Logic/AnalyticsLogic.swift

```diff
@@ -199,11 +300,76 @@ extension Logic.Analytics {
   }
 
   /// Updates the opportunity window for the event without exposure
-  struct UpdateEventWithoutExposureOppurtunityWindow: AppStateUpdater {
+  struct SetEventWithoutExposureOpportunityWindow: AppStateUpdater {
     let window: AnalyticsState.OpportunityWindow
 
     func updateState(_ state: inout AppState) {
       state.analytics.eventWithoutExposureWindow = self.window
     }
   }
+
+  /// Sets the opportunity window for the dummy analytics traffic
+  struct SetDummyTrafficOpportunityWindow: AppStateUpdater {
+    let dummyTrafficStochasticDelay: Double
+    let now: Date
+
+    func updateState(_ state: inout AppState) {
+      let windowStart = self.now.addingTimeInterval(self.dummyTrafficStochasticDelay)
+      let windowDuration = AnalyticsState.OpportunityWindow.secondsInDay
+      state.analytics.dummyTrafficOpportunityWindow = .init(windowStart: windowStart, windowDuration: windowDuration)
+    }
+  }
+}
+
+// MARK: Helpers
+
+private extension AnalyticsRequest.Body {
+  /// Creates a dummy request
+  static func dummy(deviceTokenLength: Int) -> Self {
```

Comment:
why not as a special `init` of `Body`?

Reply:
I find it more explicit to have it as a static constructor. It gives a name (`dummy`) to the type of initialization we expect.

Reply:
should we move the static method in the main struct? At least we can test it

Reply:
Done